### ISSUE TITLE
Question: how does memory replay backprogagation work with multiple models in series

### ISSUE THREAD

**Issue by pfeatherstone** [lucidrains/recurrent-memory-transformer-pytorch#13]
Say we have 2 instances of `RecurrentMemoryTransformer` one after the other. The first returns embeddings, not logits, which are fed into the second instance which is a continuous (without the embedding layer) version of `RecurrentMemoryTransformer`. Suppose we have a loss function which uses the outputs of the second model and the input of the first model. How does memory replay backpropagation work with this system?

**Comment by lucidrains** [comment]
haha, i haven't worked this out either. i can tell you for sure that no one is going to follow up on MRBP, since it isn't some hot paper, so you are kind of on your own

**Comment by lucidrains** [comment]
@pfeatherstone but they did replace lstms

you should have seen the monstrous circuits being made with lstms before transformers came along

**Comment by pfeatherstone** [comment]
Transformers on their own don't solve recurrence though...

**Comment by lucidrains** [comment]
@pfeatherstone yea i agree recurrence isn't going away

**Comment by pfeatherstone** [comment]
Have you had a chance to look at RWKV ?

**Comment by lucidrains** [comment]
@pfeatherstone yea i have, but i think at best linear attention can only come close to full attention, and not exceed it

funny enough, in a [recent linear attention paper](https://arxiv.org/abs/2307.14995), they ended up using the flash attention formulation...

**Comment by lucidrains** [comment]
there is also the limitations section in the rwkv paper. have you seen that?

**Comment by lucidrains** [comment]
anyways, i keep an eye on it, but not something i believe enough to go all-in on
